src/Savestate.cpp

```diff
@@ -283,10 +308,32 @@ void Savestate::VarArray(void* data, u32 len)
 
     if (Saving)
     {
-        fwrite(data, len, 1, file);
+        if (mStream)
```

Comment:
you can remove all fwrite and fread stuff here, always work on a memory stream and perform one big I/O operation at the start/end. It's cleaner and will probably be faster too because it doesn't constantly call into FS functions.